### PR TITLE
AURORA: TheWitcherSave writer

### PR DESCRIPTION
**Pull request by Nostritius**
In the process of writing the unit test for the TheWitcherSaveFile class, i wrote also a writer for TheWitcherSave files. This PR contains this ready made writer plus a unit test for it.

### CODE REVIEW

File: src/aurora/thewitchersavewriter.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Writer for writing TheWitcherSave files.
+ */
+
+#include "src/common/encoding.h"
+
+#include "thewitchersavewriter.h"
```
Comment (DrMcCoy) [inline src/aurora/thewitchersavewriter.cpp]: Full paths, please

File: src/aurora/thewitchersavewriter.cpp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+namespace Aurora {
+
+static const uint32 kRGMHID = MKTAG('R', 'G', 'M', 'H');
+
+TheWitcherSaveWriter::TheWitcherSaveWriter(const Common::UString &areaName, Common::SeekableWriteStream &stream) :
+		_stream(stream), _finished(false) {
+	// Write the magic id.
```
Comment (DrMcCoy) [inline src/aurora/thewitchersavewriter.cpp]: No period at the end of normal single-line comments

File: src/aurora/thewitchersavewriter.cpp [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	_resources.push_back(resource);
+}
+
+void TheWitcherSaveWriter::finish() {
+	if (_finished)
+		throw Common::Exception("TheWitcherSave::finish() Archive is already finished");
+
+	size_t resourceTableOffset = _stream.pos();
+
+	for (int i = 0; i < _resources.size(); ++i) {
```
Comment (DrMcCoy) [inline src/aurora/thewitchersavewriter.cpp]: size_t

File: src/aurora/thewitchersavewriter.cpp [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	resource.size = _stream.writeStream(stream);
+
+	_resources.push_back(resource);
+}
+
+void TheWitcherSaveWriter::finish() {
+	if (_finished)
+		throw Common::Exception("TheWitcherSave::finish() Archive is already finished");
+
+	size_t resourceTableOffset = _stream.pos();
```
Comment (DrMcCoy) [inline src/aurora/thewitchersavewriter.cpp]: const

File: src/aurora/thewitchersavewriter.cpp [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+	for (int i = 0; i < _resources.size(); ++i) {
+		Resource r = _resources[i];
```
Comment (DrMcCoy) [inline src/aurora/thewitchersavewriter.cpp]: const &

File: src/aurora/thewitchersavewriter.h [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+#ifndef AURORA_THEWITCHERSAVEWRITER_H
+#define AURORA_THEWITCHERSAVEWRITER_H
+
+#include "src/common/writestream.h"
+
+namespace Aurora {
+
+class TheWitcherSaveWriter {
+public:
+	/**
```
Comment (DrMcCoy) [inline src/aurora/thewitchersavewriter.h]: Doxygen comments should start with the summary in the first line with the `/**`

**Comment by Nostritius** [comment]
The issues  should now be fixed

**Comment by DrMcCoy** [comment]
Merged as 43b3d8f67...9d1fb8b03, thanks! :)

**Pull request closed without merging**